pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -384,6 +405,140 @@ func (r *ReconcileComplianceScan) phaseLaunchingHandler(h scanTypeHandler, logge
 	return reconcile.Result{}, nil
 }
 
+// getRuntimeKubeletConfig gets the runtime kubeletconfig for a node
+// by fetching /api/v1/nodes/$nodeName/proxy/configz endpoint use the kubeClientset
+// and store it in a configmap for each node
+func (r *ReconcileComplianceScan) getRuntimeKubeletConfig(nodeName string) (string, error) {
+	// get the runtime kubeletconfig
+	kubeletConfigIO, err := r.ClientSet.CoreV1().RESTClient().Get().RequestURI("/api/v1/nodes/" + nodeName + "/proxy/configz").Stream(context.TODO())
+	if err != nil {
+		return "", err
+	}
+	defer kubeletConfigIO.Close()
+	kubeletConfig, err := ioutil.ReadAll(kubeletConfigIO)
+	if err != nil {
+		return "", err
+	}
+	// kubeletConfig is a byte array, we need to convert it to string
+	kubeletConfigStr := string(kubeletConfig)
+	return kubeletConfigStr, nil
+}
+
+func (r *ReconcileComplianceScan) createKubeletConfigCM(instance *compv1alpha1.ComplianceScan, node *corev1.Node) error {
+	kubeletConfig, err := r.getRuntimeKubeletConfig(node.Name)
+	if err != nil {
+		return fmt.Errorf("cannot get the runtime kubelet config for node %s: %v", node.Name, err)
+	}
+	// create a configmap for the node
+	kubeletConfigCM := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getKubeletCMNameForScan(instance, node),
```

Comment:
We're doing this for each node + scan permutation. This means that we could be storing multiple versions of the same kubelet runtime config as ConfigMaps if multiple scans are running at once.

If we want to optimize this so each scan uses the same kubelet runtime config, and ultimately the same config map, we'd need to make sure we create them gracefully. For example, if two scans run at the same time and attempt to create the same config map for the same node, we could hit a race condition.

If we decide to go that route, we might want to consider a semaphore here to protect the config map from being mutated at the same time.